invenio_records_resources/services/base/utils.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Services utils."""
+
+
+def map_search_params(service_search_config, params):
+    """Map search params to a dictionary, useful for searches in DB.
+
+    :params service_search_config: search configuration class
+    :params params: the user query params for the search
+    :returns dict: a dict in this shape:
+        {
+            "page": int,
+            "size": int,
+            "sort": list of fields for sorting,
+            "sort_direction": string,
+            "q": string,
+        }
+    """
+    _config = service_search_config
+
+    page = params.get("page", 1)
+    size = params.get(
+        "size",
+        _config.pagination_options.get("default_results_per_page"),
+    )
+
+    _sort_name = (
+        params.get("sort")
+        if params.get("sort") in _config.sort_options
+        else _config.sort_default
+    )
+    _sort_direction_name = (
+        params.get("sort_direction")
+        if params.get("sort_direction") in _config.sort_direction_options
+        else _config.sort_direction_default
+    )
+
+    sort = _config.sort_options.get(_sort_name)
+    sort_direction = _config.sort_direction_options.get(_sort_direction_name)
+
+    query_params = params.get("q", "")
+
+    return {
+        "page": page,
+        "size": size,
+        "sort": sort.get("fields"),
```

Comment:
it looks like that it will return a `list` of fields when sort is passed, but a `string` when default.

Reply:
They are both a `list`: with `?sort=message` param it looks like this `{'page': 1, 'size': 25, 'sort': ['message'], 'sort_direction': <function asc at 0x104e42c10>, 'q': ''}` and without any param like this: `{'page': 1, 'size': 25, 'sort': ['created'], 'sort_direction': <function asc at 0x104e42c10>, 'q': ''}`